packages/cli/src/commands/integration/remove.ts

```diff
@@ -0,0 +1,357 @@
+    return 0;
+  }
+
+  // Uninstall Integration
+  try {
+    client.output.spinner('Uninstalling integration…', 1000);
+    await removeIntegration(client, integrationConfiguration, team);
+  } catch (error) {
+    client.output.error(
+      chalk.red(
+        `Failed to remove ${chalk.bold(integrationName)}: ${(error as Error).message}`
+      )
+    );
+    return 1;
+  }
+
+  client.output.success(`${chalk.bold(integrationName)} successfully removed.`);
+  return 0;
+}
+
+async function handleUnlinkOrDeleteResource(
+  client: Client,
+  team: Team,
+  resourceName: string,
+  args: string[],
+  flags: RemoveCommandFlags
+): Promise<number> {
+  client.output.spinner('Retrieving resource…', 500);
+  const resources = await getResources(client, team.id);
+  const targetedResource = resources.find(
+    resource => resource.name === resourceName
+  );
+
+  if (!targetedResource) {
+    client.output.error(`No resource ${chalk.bold(resourceName)} found.`);
+    return 0;
+  }
+
+  if (args.length === 3) {
+    if (flags['--unlink-all']) {
+      client.output.error(
+        'Cannot specify a project when using the `--unlink-all` flag.'
+      );
+      return 1;
+    }
+
+    const unlinkProjectResults = await handleUnlinkProject(
+      client,
+      targetedResource,
+      args[2],
+      !!flags['--yes']
+    );
+    if (unlinkProjectResults !== undefined) {
+      return unlinkProjectResults;
+    }
+  } else if (flags['--unlink-all']) {
+    const unlinkAllResults = await handleUnlinkAllProjects(
+      client,
+      targetedResource,
+      !!flags['--yes']
+    );
+    if (unlinkAllResults !== undefined) {
+      return unlinkAllResults;
+    }
+  }
+
+  if (flags['--delete']) {
+    const deleteResult = await handleDeleteResource(
+      client,
+      team,
+      targetedResource,
+      !!flags['--yes']
+    );
+    if (deleteResult !== undefined) {
+      return deleteResult;
+    }
+  }
+
+  return 0;
+}
+
+// $ vc integration remove [resource-name] [project]
+async function handleUnlinkProject(
+  client: Client,
+  resource: Resource,
+  projectName: string,
+  skipConfirmation: boolean
+): Promise<number | undefined> {
+  const project = resource.projectsMetadata?.find(
+    project => projectName === project.name
+  );
+  if (!project) {
+    client.output.log(
+      `Could not find project ${chalk.bold(projectName)} linked to resource ${chalk.bold(resource.name)}.`
+    );
+    return 0;
+  }
+
+  if (
+    !skipConfirmation &&
+    !(await confirmUnlinkProject(client, resource, project))
+  ) {
+    client.output.log('Canceled');
+    return 0;
+  }
+
+  try {
+    client.output.spinner('Unlinking resource…', 500);
+    await disconnectResourceFromProject(client, resource, project);
+    client.output.success(
+      `Unlinked ${chalk.bold(project.name)} from ${chalk.bold(resource.name)}`
+    );
+    resource.projectsMetadata = resource.projectsMetadata?.filter(
+      project => projectName !== project.name
+    );
```

Comment:
This is an intentional side-effect. The `--delete` flag checks for linked projects on the object because a resource with connected projects cannot be deleted. The act of unlinking a project from a resource, in this situation, is updating that object to reflect the new reality.

Assuming this is not desired, the alternative would be following immutability rules & cloning the resource then returning it. This breaks the other return pattern though, so failures would need to be switched to error throws.